### PR TITLE
[Android] Implement pending focus management logic in Handlers

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement pending focus management logic in Handlers. Changes in:
- CheckBox
- DatePicker
- Entry
- Editor
- Picker
- RadioButton
- TimePicker

We didn't bring over a lot of the auto focus changes from XF because those need to be validated as correct. Typically you never want to modify the focus of the platform for the sake of it. 

The behavior in android was modified to match iOS with how `Pickers` will unfocus when you click `Done` but that's not how WinUI works.

Removed the code that closes the keyboard when you click the backbutton because that's handled by the platform

Removed the code that closes the keyboard when unfocusing because that's not what the platform does by default and we don't want to modify the platform default without a very very good reason.  Focus/keyboard management is very broken in `Forms`

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
Updated several tasks in https://github.com/dotnet/maui/issues/4808

File: src/Core/src/Handlers/Editor/EditorHandler.Android.cs [hunk 1]
```diff
@@ -95,5 +106,11 @@ void OnFocusedChange(object? sender, FocusChangeEventArgs e)
 			if (!e.HasFocus)
 				VirtualView?.Completed();
 		}
+
+		void OnKeyboardBackPressed(object? sender, EventArgs eventArgs)
+		{
+			VirtualView?.Completed();
+			PlatformView?.ClearFocus();
```
Comment (PureWeen) [inline src/Core/src/Handlers/Editor/EditorHandler.Android.cs:113]: I don't think we want to influence the focus ourselves here. I'm guessing this code was fixing a probably incorrectly in XF

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Editor/EditorHandler.Android.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/Editor/EditorHandler.Android.cs]: Updated.

**Comment by Redth** [comment]
Fixes #3380

**Review by PureWeen — APPROVED** [review]

**Comment by nzcoward** [comment]
#5321 was marked as a dupe of #3380, and if you've closed #3380, should we expect #5321 to be fixed? in 6.0.400 the ReturnType Next still does not work on Android.

Or have these changes not made their way into a release yet?

**Comment by supershopping** [comment]
Any update on this? When will this fix to be released?

**Comment by dangernme** [comment]
This is still a problem in VS Release 17.4.1. This is a very basic use case and I really have to consider to "unfocus" .net MAUI to be productive.

**Comment by UchiTesting** [comment]
I just reopened a personal project on MAUI and I still observe this behaviour.
I need to press the back button from the phone in order to have this close.

**Pull request merged**